watcher/lib/grafana/grafana.go

```diff
@@ -93,16 +93,18 @@ type CreateDashboardRequest struct {
 	Overwrite bool `json:"overwrite"`
 }
 
-// DeleteDashboard deletes dashboard
-// dashboard name is obtained from dashboard data field 'title'
+// DeleteDashboard deletes a dashboard specified with data.
+// data is expected to be JSON-encoded and contain a field named `title` which names the dashboard to delete.
 func (c *Client) DeleteDashboard(data string) error {
-	// dashboard data should be a valid JSON
-	var dashboardJSON map[string]interface{}
+	type header struct {
+		Title string `json:"title"`
+	}
+	var dashboardJSON header
```

Comment:
Could be simpler to just:
```go
var dashboardJSON struct {
	Title string `json:"title"`
}
```
I think it's my bad to not have suggested this in the first place.